Src/Newtonsoft.Json/Serialization/JsonArrayContract.cs

```diff
@@ -207,11 +212,7 @@ public JsonArrayContract(Type underlyingType)
                 _parameterizedConstructor = CollectionUtils.ResolveEnumerableCollectionConstructor(underlyingType, CollectionItemType);
 
 #if HAVE_FSHARP_TYPES
-                if (!HasParameterizedCreatorInternal && underlyingType.Name == FSharpUtils.FSharpListTypeName)
```

Comment:
Yes, I extracted the logic in a new method and call it from here and from the other branch.